### ISSUE TITLE
How does hammer model divergence+synchronization with the DUT

### ISSUE THREAD

**Issue by jerryz123** [rivosinc/hammer#3]
A lock-step cosimulation framework would typically need to support allowable architectural divergences between the functional model and the DUT. How is that intended to work with Hammer's interface?

Say, for interrupts, how are DUT and simulator interrupts intended to be synchronized?

**Comment by jerinjoy** [comment]
Yes this is something that we want to add. We haven't worked out the details yet.

**Comment by jerryz123** [comment]
Great! I'm looking forward to it. 
I'm also happy to help with anything, either in Hammer, Spike, or testing the cosim with open RTL.
